lightning/src/ln/onchaintx.rs

```diff
@@ -318,20 +336,18 @@ impl<ChannelSigner: Sign> OnchainTxHandler<ChannelSigner> {
 		}
 
 		writer.write_all(&byte_utils::be64_to_array(self.onchain_events_waiting_threshold_conf.len() as u64))?;
-		for (ref target, ref events) in self.onchain_events_waiting_threshold_conf.iter() {
```

Comment:
I was working off the understanding that its ok to break the serialization format for now. But maybe that's no longer the case?

As for your question, any such `ChannelMonitor` could have been serialized with such events, so I would think that deserialization would fail in that case. They'd have to use the old code to deserialize, clear the events without producing any new ones, and then re-serialize. Then it would work with the new code.

Reply:
Right, given a monitor with no such events, it should roundtrip old code -> new code fine.